fs/parsetime.go

```diff
@@ -0,0 +1,106 @@
+package fs
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// Time is a time.Time with some more parsing options
+type Time time.Time
+
+// For overriding in unittests.
+var (
+	timeNowFunc = func() time.Time { return time.Now() }
+)
+
+// Turn Time into a string
+func (t Time) String() string {
+	if !t.IsSet() {
+		return "off"
+	}
+	return time.Time(t).Format(time.RFC3339Nano)
+}
+
+// IsSet returns if the time is not zero
+func (t Time) IsSet() bool {
+	return !time.Time(t).IsZero()
+}
+
+// parse the date as time in various date formats
+func parseTimeDates(date string) (t time.Time, err error) {
+	var instant time.Time
+	for _, timeFormat := range timeFormats {
+		instant, err = time.Parse(timeFormat, date)
+		if err == nil {
+			return instant, nil
+		}
+	}
+	return t, err
+}
+
+// ParseTime parses a time or duration string as a Time.
+func ParseTime(date string) (t time.Time, err error) {
+	if date == "off" {
+		return time.Time{}, nil
+	}
+
+	now := timeNowFunc()
+
+	// Attempt to parse as unix timestamp
+	unixMillis, err := strconv.ParseInt(date, 10, 64)
+	if err == nil && unixMillis >= 0 {
+		return time.UnixMilli(unixMillis).UTC(), nil
+	}
```

Comment:
Do we really need this? It isn't a good date format!

Reply:
The main reason I added this is because of the JSON unmarshalling being there, and whenever I encounter timestamps in JSON they are unix millis (`Date.now()` in javascript). But I'm not sure why the json unmarshalling is there, but I copied it from `fs.Duration`. Either way I'd be fine removing it, I don't see a real need for it otherwise.

How should I implement `UnmarshalJSONFlag`'s `setInt` method if not a unix milli? Just raise an error?

Reply:
Ah I see. The JSON parsing of fs.Duration as an integer is because it is returned as an integer number of Nanoseconds so needs to be parsed that way too. I think we can do without the integer parsing here, so I think we should remove this.

Reply:
Alright, I've removed UnixMilli parsing in https://github.com/rclone/rclone/pull/6035/commits/e57cebdacb86ab2866e81811cad9458690bd2ac4